Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerAttachmentsView.swift

```diff
@@ -5,91 +5,163 @@
 import StreamChat
 import UIKit
 
-open class MessageComposerAttachmentsView<ExtraData: ExtraDataTypes>: UIView,
+open class MessageComposerAttachmentsView<ExtraData: ExtraDataTypes>: View,
+    UIConfigProvider,
     UICollectionViewDelegate,
     UICollectionViewDataSource {
+    // MARK: - Underlying types
+    
+    public enum State {
+        case documents([(preview: UIImage, name: String, size: Int64)])
+        case media(previews: [UIImage])
+    }
```

Comment:
Wouldn't be our life easier if we have two distinct types instead of switches all over this class? 🤔

Reply:
Yeah, I can admit I'm a fan of `states-and-switches` and it makes sense here to have 2 separate types 😄